semantic-rust/src/Language/Rust/Tags.hs

```diff
@@ -50,6 +59,172 @@ gtags
   -> m ()
 gtags = traverse1_ @ToTags (const (pure ())) tags
 
+-- enum SyntaxType {
+--   FUNCTION = 0;
+--   METHOD = 1;
+--   CLASS = 2;
+--   MODULE = 3;
+--   CALL = 4;
+--   TYPE = 5;
+--   INTERFACE = 6;
+--   IMPLEMENTATION = 7;
+-- }
+
+yieldTag :: (Has (Reader Source) sig m, Has (Writer Tags.Tags) sig m) => Text -> P.SyntaxType -> P.NodeType -> Loc -> Range -> m ()
+yieldTag name kind ty loc srcLineRange = do
+  src <- ask @Source
+  Tags.yield (Tag name kind ty loc (Tags.firstLine src srcLineRange) Nothing)
+
+
+instance ToTags Rust.ModItem where 
+  tags 
+    t@Rust.ModItem 
+      {
+        ann = loc@Loc {byteRange},
+        body,
+        name = Parse.Success (Rust.Identifier {text, ann})
+      } = do 
+        case body of 
+          Just (Parse.Success decls) -> gtags decls
+          _ -> pure ()
+        where yield name ann = yieldTag name P.FUNCTION P.DEFINITION ann byteRange >> gtags t
+  tags _ = pure ()
+
+
+instance ToTags Rust.FunctionItem where
+  tags
+    t@Rust.FunctionItem
+      { 
+        ann = loc@Loc {byteRange},
+        returnType,
+        body = Parse.Success block,
+        name,
+        parameters = Parse.Success parameters,
+        typeParameters
+      } = do
+      case returnType of
+        Just (Parse.Success type') -> gtags type'
+        _ -> pure ()
+      case typeParameters of
+        Just (Parse.Success typeParams) -> gtags typeParams
+        _ -> pure ()
+      case name of
+        EPrj (Rust.Identifier {text, ann}) -> yield text ann
+        EPrj (Rust.Metavariable {text, ann}) -> yield text ann
+        _ -> pure ()
+      gtags block
+      where
+        yield name ann = yieldTag name P.FUNCTION P.DEFINITION ann byteRange >> gtags t
+  tags _ = pure ()
+
+
+instance ToTags Rust.CallExpression where
+  tags
+    t@Rust.CallExpression 
+      {
+        ann = loc@Loc {byteRange},
+        function = Parse.Success expr,
+        arguments = Parse.Success args
+      } = gtags expr
+  tags _ = pure ()
```

Comment:
Here, too, I think we can just call `gtags t`—or, since that’s equivalent to the default instance, drop it entirely.